tools/release_builder.py

```diff
@@ -491,10 +490,37 @@ def _update_name_and_version(self, pkgdir):
             f.write('\n')
 
 
+    def _update_marathon_json(self, pkgdir):
+        '''Updates the marathon.json definition to contain the desired name and version strings.
+        '''
+        # note: the file isn't valid JSON, so we edit the raw content instead
+        marathon_file_name = os.path.join(pkgdir, 'marathon.json.mustache')
+        with open(marathon_file_name) as f:
+            orig_marathon_lines = f.readlines()
+
+        marathon_lines = []
+        for line in orig_marathon_lines:
+            name_match = re.match(r'^ *"PACKAGE_NAME": ?"(.*)",?$', line.rstrip('\n'))
```

Comment:
We could use `re.sub()` to perform the substitution directly. Also, is the `rstrip()` on the line strictly required?

Reply:
`line` ends with a trailing newline. I've had wonky behavior with passing newlines through regexes before.

Reply:
Tried `re.sub()`:
- Pattern to be used is significantly more spaghetti-like: `re.sub(r'^( *"{}": *")(.*)(",?)$'.format(name), r'\1{}\3'.format(val), line)`
- Regexp apparently doesn't work on certain input, for reasons that I can't figure out:
```
input: '    "PACKAGE_VERSION": "stub-universe",
'
Traceback (most recent call last):
  File "/usr/lib/python3.5/sre_parse.py", line 938, in expand_template
    literals[index] = g(group) or empty
IndexError: no such group

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./tools/release_builder.py", line 583, in <module>
    sys.exit(main(sys.argv))
  File "./tools/release_builder.py", line 564, in main
    response = builder.release_package()
  File "./tools/release_builder.py", line 526, in release_package
    self._update_marathon_json(pkgdir)
  File "./tools/release_builder.py", line 511, in _update_marathon_json
    line = with_updated_envvar(line, "PACKAGE_VERSION", self._pkg_version)
  File "./tools/release_builder.py", line 506, in with_updated_envvar
    newline = re.sub(r'^( *"{}": *")(.*)(",?)$'.format(name), r'\1{}\3'.format(val), line)
  File "/usr/lib/python3.5/re.py", line 182, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "/usr/lib/python3.5/re.py", line 330, in filter
    return sre_parse.expand_template(template, match)
  File "/usr/lib/python3.5/sre_parse.py", line 940, in expand_template
    raise error("invalid group reference")
sre_constants.error: invalid group reference
```

Going to stick with `re.match` + `str.replace` until someone who's better at regexes can figure out the correct permutation to use here.

Reply:
Yes, marathon.json.mustache typically has things like conditional `{{#foo}}` that will cause the JSON to not parse